client/src/Auth/TabPanel.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from "react";
+import Typography from "@material-ui/core/Typography";
+import TextField from "@material-ui/core/TextField";
+import { Button } from "@material-ui/core";
+
+import { useStyles } from "./Auth.components";
+
+interface Props {
+  activeTab: number;
+  index: number;
+}
+
+const TabPanel: React.FC<Props> = ({ activeTab, index }) => {
+  const classes = useStyles();
+
+  const [credentials, setCredentials] = useState({
+    fullName: "",
+    username: "",
+    password: "",
+    confirmPassword: ""
+  });
+
+  const [error, setError] = useState("");
+
+  const fullName = credentials.fullName;
+  const username = credentials.username;
+  const password = credentials.password;
+  const confirmPassword = credentials.confirmPassword;
+
+  const isValidUsername = !!username && !/^[0-9a-zA-Z_.-]+$/.test(username);
+  const isValidPassword = !!password && password.length < 5;
+  const isValidConfirmedPassword =
+    !!confirmPassword && confirmPassword !== password;
+
+  const handleChange = (prop: string) => (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setCredentials({ ...credentials, [prop]: event.target.value });
+  };
```

Comment:
```
const handleChange = ({target: {name, value}}) => {
   setCredentials({...credentials, [name]: value]
}
```

No need to create a function returning a function when you can store the field name in JSX